File: backend/src/main/kotlin/dataType/Rent.kt
package dataType

import kotlinx.serialization.Serializable

@Serializable
data class Rent(
    val id: String,
    val idUser: String,
    val idItem: String,
    val idStorage: String,
    val dateFrom: String,
    val dateUntil: String,
    val dateStatus: String,
    val status: String
)

File: frontend/src/dataclasses/rent.ts
type Rent = Id & {
  idUser: string
  idItem: string
  idStorage: string
  dateFrom: string
  dateUntil: string
  dateStatus: string
  status: string
}

export default Rent

export const newRent = (
  id: string = "",
  idUser: string = "",
  idStorage: string = "",
  dateFrom: string = "",
  dateUntil: string = "",
  dateStatus: string = "",
  status: string = ""
) => {
  const retval = {} as Rent
  retval.id = id
  retval.idUser = idUser
  retval.idStorage = idStorage
  retval.dateFrom = dateFrom
  retval.dateUntil = dateUntil
  retval.dateStatus = dateStatus
  retval.status = status
  return retval
}
